src/main/java/org/broadinstitute/hellbender/utils/haplotype/EventMap.java

```diff
@@ -314,13 +314,18 @@ public static TreeSet<Integer> buildEventMapsForHaplotypes( final List<Haplotype
 
         if( debug ) logger.info("=== Best Haplotypes ===");
         for( final Haplotype h : haplotypes ) {
-            // Walk along the alignment and turn any difference from the reference into an event
-            h.setEventMap(new EventMap(h, ref, refLoc, "HC" + hapNumber++, maxMnpDistance));
+            if (h instanceof PartiallyDeterminedHaplotype) {
+                // Since PD haplotypes Know what allels are variants, simply ask it and generate the map that way.
+                h.setEventMap(new EventMap(((PartiallyDeterminedHaplotype) h).getDeterminedAlleles()));
```

Comment:
I'm giong to punt on this simply because the method setEventMap() is covered by a a lot of tests and it would be difficult to refactor. I agree with you and I'll add a todo.